internal/pkg/plan/delete-template/plan_test.go

```diff
@@ -0,0 +1,142 @@
+	d := dependencies.NewTestContainer()
+	d.SetFs(fs)
+
+	// Mocked API response
+	d.UseMockedSchedulerApi()
+	_, httpTransport := d.UseMockedStorageApi()
+	branches := []*model.Branch{{BranchKey: model.BranchKey{Id: 123}, Name: "Main", IsDefault: true}}
+	httpTransport.RegisterResponder(
+		"GET", `=~/storage/dev-branches`,
+		httpmock.NewJsonResponderOrPanic(200, branches),
+	)
+	httpTransport.RegisterResponder(
+		"GET", `=~/storage/branch/123/metadata`,
+		httpmock.NewJsonResponderOrPanic(200, []storageapi.Metadata{
+			{
+				Id:        "1",
+				Key:       "KBC.KAC.templates.instances",
+				Value:     `[{"instanceId":"inst1","instanceName":"Instance 1","templateId":"tmpl1","repositoryName":"repo","version":"1.0.0","created":{"date":"%s","tokenId":"12345"},"updated":{"date":"%s","tokenId":"12345"}}]`,
+				Timestamp: "",
+			},
+		}),
+	)
+	getGenericExResponder, err := httpmock.NewJsonResponder(200, map[string]interface{}{
+		"id":   "ex-generic-v2",
+		"type": "extractor",
+		"name": "Generic",
+	})
+	assert.NoError(t, err)
+	httpTransport.RegisterResponder("GET", `=~/storage/components/ex-generic-v2`, getGenericExResponder.Times(3))
+	httpTransport.RegisterResponder(
+		"GET", `=~/storage/branch/123/components`,
+		httpmock.NewJsonResponderOrPanic(200, []*model.ComponentWithConfigs{
+			{
+				Component: &model.Component{ComponentKey: model.ComponentKey{Id: `ex-generic`}, Name: `Generic`, Type: `extractor`},
+				Configs: []*model.ConfigWithRows{
+					{Config: &model.Config{ConfigKey: model.ConfigKey{Id: "123"}, Name: `Config 1`, Content: orderedmap.New()}},
+					{Config: &model.Config{ConfigKey: model.ConfigKey{Id: "456"}, Name: `Config 2`, Content: orderedmap.New()}},
+					{Config: &model.Config{ConfigKey: model.ConfigKey{Id: "789"}, Name: `Config 3`, Content: orderedmap.New()}},
+				},
+			},
+		}),
+	)
+	httpTransport.RegisterResponder(
+		"GET", `=~/storage/branch/123/search/component-configurations`,
+		httpmock.NewJsonResponderOrPanic(200, storageapi.ConfigMetadataResponse{
+			{
+				ComponentId: "ex-generic-v2",
+				ConfigId:    "123",
+				Metadata: []storageapi.Metadata{
+					{
+						Id:        "",
+						Key:       "KBC.KAC.templates.instanceId",
+						Value:     "inst1",
+						Timestamp: "",
+					},
+				},
+			},
+			{
+				ComponentId: "ex-generic-v2",
+				ConfigId:    "456",
+				Metadata: []storageapi.Metadata{
+					{
+						Id:        "",
+						Key:       "KBC.KAC.templates.instanceId",
+						Value:     "inst2",
+						Timestamp: "",
+					},
+				},
+			},
+			{
+				ComponentId: "ex-generic-v2",
+				ConfigId:    "789",
+				Metadata: []storageapi.Metadata{
+					{
+						Id:        "",
+						Key:       "KBC.KAC.templates.instanceId",
+						Value:     "inst1",
+						Timestamp: "",
+					},
+				},
+			},
+		}),
+	)
+
+	// Load state
+	prj, err := d.LocalProject(false)
+	assert.NoError(t, err)
+	projectState, err := prj.LoadState(loadState.Options{LoadLocalState: true, LoadRemoteState: true})
```

Comment:
@webrouse Asi jsem se do toho dost zamotal a potřeboval bych poradit. 🙂 Vytvořil jsem pro to mazání konfigů `Plan` aby to bylo možný pustit s `DryRun` a opisuju to v podstatě podle `Rename` plánu včetně testů. 

Když ty `all` konfigy do searche dostávám přes `b.State.RemoteObjects()` https://github.com/keboola/keboola-as-code/blob/9320c731b4d4e51859051404bf7864769a99d8c9/internal/pkg/plan/delete-template/delete.go#L27
tak to znamená že tady ten `prj.LoadState()` musím zavolat s `LoadRemoteState: true`, jo? To chce hromadu mockování API a stejně mi to pořád nějak failuje, tak se raději ptám jestli to nedělám blbě a zbytečně. 🙂

Reply:
Dnes sa tomu asi uz nestihmen venovat :/ mozeme to prebrat zajtra, ...

Reply:
V pohodě, musím se stejně chystat na sprint review.